src/keys.rs

```diff
@@ -567,15 +567,34 @@ impl SharedSecret {
         self.0.to_bytes()
     }
 
+    /// Only for use in batched note encryption.
+    pub(crate) fn batch_to_affine(shared_secrets: &[Option<Self>]) -> Vec<pallas::Affine> {
```

Comment:
I'll see if I can make this return an `impl Iterator`, so I can move the interpolation logic into here for clarity.